docs(progress-indicator): add progress-indicator (#33)
siemens/ix-docs: docs/components/progress-indicator/guide.md

---
doc-type: 'tab-item'
---

import DoAndDont from '@site/src/components/DoAndDont';

# Progress indicator - Usage

Progress indicators inform users about the status of ongoing processes, e.g. loading data, submitting forms or processing non-blocking operations.

> 

Reviewer: goood point!

![Progress indicator anatomy](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=2094-345&t=pq3AmdWOVOjIx4S4-4)

1. Label
2. Helper text
3. Fill
4. Track
5. Content slot

The basic anatomy of progress indicators consists of a value indicator moving along a background track. The track represents the total expected length while the fill shows the current progress.

## Variants

Progress indicator variants (types):
- **Linear:** Use for horizontal layouts or when the progress value should be better visible.
- **Circular:** Use for compact or centered layouts, e.g. when it is replacing an icon. 

## Options

- **Status:** Use the status to reflect the outcome or current condition of a process:
	- Default: Normal, ongoing process without special attention needed
	- Success: Process completed successfully

> 	- Info: Process ongoing with low-impact additional information (e.g. "Compressing files before upload")

Reviewer 2: ```suggestion
	- Info: Process ongoing with low-impact additional information, e.g. "Compressing files before upload".
```

> 	- Info: Process ongoing with low-impact additional information (e.g. "Compressing files before upload")

Reviewer: somehow this was added twice, so I'll just resolve this one :)

	- Warning: Process ongoing but needs attention (e.g. "Storage space is running low")
	- Error: Process interrupted or failed (e.g. due to network connection problems)
	- Paused: Process temporarily stopped by the system or by the user (e.g. by pressing a button)

> - **Alignment:** By default, the label, control and helper text are left-aligned. Use the centered option to accommodate layouts with vertical reading patterns e.g. low-width containers.

Reviewer 2: ```suggestion
- **Alignment:** By default, the label, control and helper text are left-aligned. Use the centered option to accommodate layouts with vertical reading patterns, e.g. low-width containers.
```

- **Value, min and max:** The progress range is customizable. By default, it spans from 0 to 100. Adjust the min and max values to suit the specific use case.
- **Content slot:** Use this slot to display additional content such as percentage values or custom elements like icons.

:::info
For more information about correct copywriting for helper text or label please see our [UX Writing guidelines](../../guidelines/language/basics.md).
:::
## Behavior in context

- **Resizing:** The total width of progress indicators is `24rem` by default. Customize it appropriately for its context.
- **Text overflow:** If the helper text or label are too long for the available horizontal space, it has a line-break.
- **Slot:** Slots in linear progress indicators have a min-width of `2.25rem` to accommodate using values from 0% to 100% without changing the width of the actual bar. 

![progress indicator resizing](https://www.figma.com/design/wEptRgAezDU1z80Cn3eZ0o/iX-Documentation-illustrations?node-id=5650-16162&t=pq3AmdWOVOjIx4S4-4)

## Dos and Don’ts

<DoAndDont>
  <DoAndDont.Do>
    <DoAndDont.Item>Do use progress indicators consistently for similar processes</DoAndDont.Item>

>     <DoAndDont.Item>Do use progress indicators for determinate processes where progress can be measured (otherwise use [spinners](../spinner/index.mdx))</DoAndDont.Item>

Reviewer 3: Mentioning of the spinner for indeterminate processes is quite late in the dos and don'ts. Would be have benefit to have this earlier in the usage description so users looking for a solution for indeterminate processes don't need to read until the end.

    <DoAndDont.Item>Do use linear progress indicators in horizontal layouts and circular indicators in compact spaces or centered layouts</DoAndDont.Item> 
	<DoAndDont.Item>Do keep your slot content short, especially for the centered alignment (use helper texts and labels for lengthier content)</DoAndDont.Item>
  </DoAndDont.Do>
  <DoAndDont.Dont>
    <DoAndDont.Item>Don't use progress indicators for operations shorter than one second</DoAndDont.Item>

>     <DoAndDont.Item>Don’t stop at visual completion, always provide clear confirmation that the task is finished, e.g. a success message or the display of loaded content</DoAndDont.Item>

Reviewer 2: ```suggestion
    <DoAndDont.Item>Don’t only indicate progress completion with the indicator without clear task messages, e.g. success toasts or displaying the loaded content</DoAndDont.Item>
```

Reviewer: It is much better now, thanks!
